hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/compact/DefaultCompactWriter.java

```diff
@@ -0,0 +1,481 @@
+        FieldDescriptor fieldDefinition = checkFieldDefinition(fieldName, BOOLEAN);
+        int offsetInBits = fieldDefinition.getOffset();
+        int offsetInBytes = offsetInBits == 0 ? 0 : (offsetInBits / Byte.SIZE);
+        int writeOffset = offsetInBytes + offset;
+        int booleanOffsetWithinLastByte = offsetInBits % Byte.SIZE;
+        try {
+            out.writeBooleanBit(writeOffset, booleanOffsetWithinLastByte, value);
+        } catch (IOException e) {
+            throw illegalStateException(e);
+        }
+    }
+
+    IllegalStateException illegalStateException(IOException cause) {
+        return new IllegalStateException("IOException is not expected from BufferObjectDataOutput ", cause);
+    }
+
+    @Override
+    public void writeByte(@Nonnull String fieldName, byte value) {
+        int position = getFixedLengthFieldPosition(fieldName, BYTE);
+        try {
+            out.writeByte(position, value);
+        } catch (IOException e) {
+            throw illegalStateException(e);
+        }
+    }
+
+    @Override
+    public void writeChar(@Nonnull String fieldName, char value) {
+        int position = getFixedLengthFieldPosition(fieldName, CHAR);
+        try {
+            out.writeChar(position, value);
+        } catch (IOException e) {
+            throw illegalStateException(e);
+        }
+    }
+
+    @Override
+    public void writeDouble(@Nonnull String fieldName, double value) {
+        int position = getFixedLengthFieldPosition(fieldName, DOUBLE);
+        try {
+            out.writeDouble(position, value);
+        } catch (IOException e) {
+            throw illegalStateException(e);
+        }
+    }
+
+    @Override
+    public void writeFloat(@Nonnull String fieldName, float value) {
+        int position = getFixedLengthFieldPosition(fieldName, FLOAT);
+        try {
+            out.writeFloat(position, value);
+        } catch (IOException e) {
+            throw illegalStateException(e);
+        }
+    }
+
+    @Override
+    public void writeShort(@Nonnull String fieldName, short value) {
+        int position = getFixedLengthFieldPosition(fieldName, SHORT);
+        try {
+            out.writeShort(position, value);
+        } catch (IOException e) {
+            throw illegalStateException(e);
+        }
+    }
+
+    protected <T> void writeVariableLength(@Nonnull String fieldName, FieldType fieldType, T object, Writer<T> writer) {
+        try {
+            if (object == null) {
+                setPositionAsNull(fieldName, fieldType);
+            } else {
+                setPosition(fieldName, fieldType);
+                writer.write(out, object);
+            }
+        } catch (IOException e) {
+            throw illegalStateException(e);
+        }
+    }
+
+    @Override
+    public void writeString(@Nonnull String fieldName, String str) {
+        writeVariableLength(fieldName, UTF, str, ObjectDataOutput::writeString);
```

Comment:
We previously decided to not use the work `UTF` in the public API but prefer `string` instead, can we rename these internal enum names as well?

Reply:
This enum is not internal it is the part of the API. That is why we could not change it.